oeps/architectural-decisions/oep-XXXX-micro-frontend-domains.rst

```diff
@@ -0,0 +1,171 @@
+     - TBD
+   * - Resolution
+     - TBD
+   * - References
+     - TBD
+
+
+********
+Abstract
+********
+
+The decision to gradually implement a micro-frontend architecture for Open edX was a largely successful attempt at avoiding some limitations of the monolithic ``edx-platform`` codebase.  In particular, it allowed the project to integrate more and better features by leveraging newer frontend technology, all much faster than would have been possible otherwise.  At the same time, because features encapsulated in micro-frontends (henceforth, "MFEs") are very loosely coupled to each other, this architecture eliminates most integration bottlenecks, allowing individual teams working on any given feature to not be a lot more *productive*.
+
+However, roughly 3 years into this gradual re-platforming, several drawbacks to the chosen implentation of complete single-page app ("SPA") independence have been identified.  This OEP proposes a middle-ground based on Domain Driven Development ("DDD") and current industry best-practices, where a minor increase in integration costs is traded for large potential gains in user experience, customizability, ease of deployment, and in certain cases, even development efficiency.
+
+
+**********
+Motivation
+**********
+
+When moving to micro-frontends, a (now known to be) uncommon choice was made: to have *completely* independent SPAs, each with their separate deployment, endpoint, and different technology stack.  While the goal of developing platform features more efficiently was achieved, the price to pay was not insignificant.  Here are some critical issues that the new architecture introduced:
+
+* Inconsistency between MFEs and the legacy codebase
+
+  * UI is designed differently, resulting in poorer UX across boundaries;
+  * They're also implemented differently, which makes tooling, development, and branding that much harder than before, because both implementations have to be dealt with.
+
+* Inconsistency between the UIs of MFEs themselves
+
+  * Each MFE can use whatever paradigm was in vogue during its development, leading to differences (some minor, some major) in UX;
+  * This makes it inordinately difficult to theme components that are not common to all MFEs, such as headers and footers.
+
+* Regression in customizability
+
+  * To achieve similar customizability to the legacy codebase, MFEs need to be forked, with all the maintenance cost that that entails;
+  * The fact each MFE is fully independent means that it is harder to implement a consistent plugin interface between all of them.
+
+* Changes that affect multiple MFEs are costly
+
+  * Development effort is multiplied by the number of existing MFEs, as each change needs to be duplicated;
+  * Because no dependencies are shared, Webpack build time of any potentially shared library is multiplied by the number of existing MFEs.
+
+* Inefficient use of network bandwidth
+
+  * Dependencies have to be downloaded multiple times by the browser.
```

Comment:
In a similar vein, it may also be worth calling out that the current architecture limits our ability to take advantage of routing (e.g., with `react-router`) between MFEs. As is, to navigate from "MFE A" to "MFE B", we have to do a full page refresh, rather than more intelligent routing that avoids refreshing the whole page when navigating between MFEs.